### PR TITLE
Add some content to the styleguide page

### PR DESCRIPTION
**Pull request by zarvox**
A first pass at some of the things that have been bouncing around in my head.  There is surely more to be written (and a bunch more examples to put on the page), but this at least puts the things that I've tried to express in various code and design reviews into words somewhere.

### CODE REVIEW

**Comment by ocdtrekkie** [comment]
What is this style guide supposed to apply to? Sandstorm itself? Apps?

Also, where can this style guide be found? It seems strange it is not in /docs

**Comment by zarvox** [comment]
Sandstorm itself, though in my personal opinion the high-level guidelines are good ideas in general. :)

The rationale behind putting it in the shell is that it can make direct use of buttons and other CSS that canonically lives inside the shell without needing to be duplicated between e.g. the shell and the docs.  I've admittedly not made much in the way of examples here yet, but I do intend to provide a more detailed set of examples of things like buttons and forms and message boxes that I can point people at.

**Comment by kentonv** [comment]
Have you shown this to @neynah and does she think it looks right?

(I didn't look closely.)

**Comment by neynah** [comment]
I have yet to review it but I will take a closer look tomorrow. :D

File: shell/client/styleguide.html [hunk 1]
```diff
@@ -1,6 +1,95 @@
...
+    Use nouns for basically everything else, too, to communicate the information architecture.
+    Section headings should be nouns which describe the content of the section.  Labels should be
+    nouns which describe the content of the field.
+  </li>
+</ul>
+
+<h2>Style rules</h2>
+<ul>
+  <li>
+    Stick to a limited color pallete.  (TODO: add specific colors)
```
Comment (neynah) [inline shell/client/styleguide.html]: "pallete" => "palette"

File: shell/client/styleguide.html [hunk 2]
```diff
@@ -1,6 +1,95 @@
...
+    Use nouns for basically everything else, too, to communicate the information architecture.
+    Section headings should be nouns which describe the content of the section.  Labels should be
+    nouns which describe the content of the field.
+  </li>
+</ul>
+
+<h2>Style rules</h2>
+<ul>
+  <li>
+    Stick to a limited color palette.  (TODO: add specific colors)
```
Comment (neynah) [inline shell/client/styleguide.html]: I'm happy to document these.

File: shell/client/styleguide.html [hunk 3]
```diff
@@ -1,6 +1,95 @@
...
+  </li>
+</ul>
+
+<h2>Forms</h2>
+
+<ul>
+  <li>Use the <code>standard-form</code> class where appropriate.</li>
+  <li>
+    The primary action button of a form should be at the bottom right of the form. Secondary
+    buttons should line up to the left of that primary action button.
```
Comment (neynah) [inline shell/client/styleguide.html]: I believe we have primary, secondary, and tertiary buttons. I'll try to find an example in our existing UI.

**Comment by neynah** [comment]
I think this is fine to merge as-is although I do want to make some additions (e.g. adding a bunch of styled elements from https://github.com/sandstorm-io/sandstorm/pull/1635/files). We also document typography & general layout/grid patterns in detail. I'm happy to do that.

**Pull request merged**